discovery-provider/src/tasks/index_nethermind.py

```diff
@@ -239,6 +250,25 @@ def fetch_cid_metadata(db, entity_manager_txs):
                     ):
                         continue
 
+                    # Check if metadata blob was passed directly.
```

Comment:
these metadata types above will eventually all use a blob like below right?

> TODO make sure we are validating metadata structure for every applicable entity type.

if there was a mismatch now, indexing would ignore that tx and continue indexing right?

Reply:
> these metadata types above will eventually all use a blob like below right?

yes

> if there was a mismatch now, indexing would ignore that tx and continue indexing right?

just added the same metadata validation that we use on the responses from CNs in this commit https://github.com/AudiusProject/audius-protocol/pull/5046/commits/002c2cbdd6cd4520c793820c02a39e2ad4421533. It filters the metadata json to only include values for keys defined in each type's metadata format in `discovery-provider/src/tasks/metadata.py`, and if keys / values are missing in the metadata json, it uses the default values. 

in `index_nethermind.py` and `index.py`, it'll save the formatted metadata in the `cid_metadata` dict so the lookup will never fail downstream. If there's an error for some reason (which should just be if its actually not a blob and just a CID), it'll follow the old code path which, on error, raises a fatal exception for the entire block i believe. but this change shouldn't introduce anything that would cause a fatal exception to occur that otherwise wouldnt have occurred, unless someone passed in a malformed blob missing the "cid" or "data" fields.

in the future when only blobs are passed in and we dont have to bulk fetch from from CNs, we can just validate blobs on a tx-by-tx basis and skip problematic txs instead of having this weird dependency that could fail the whole block and i think stall indexing?

Reply:
yea these should be skipped we'll fix this next spring, thanks for calling out

Reply:
yeah... so currently someone can technically stall the network by sending a tx with an incorrect cid. from what i understand we intentionally require all cid metadata to be fetched successfully (see [here)](https://github.com/AudiusProject/audius-protocol/blob/964b81a54a0dfb498f728cab67693a5938ea5801/discovery-provider/src/utils/cid_metadata_client.py#L267) to prevent drift -- in case one node is actually able to retrieve and index a cid, all the nodes should be able to.

but that reasoning doesn't apply here since no network requests are involved and we should totally eliminate this stall risk if possible... after thinking about it some more, its actually p easy to add validations and skip the tx even with the bulk processing function structure. ill put up a PR in a bit. ty for the callout!